composition-js/src/__tests__/compose.test.ts

```diff
@@ -4228,4 +4228,313 @@ describe('composition', () => {
+            id: ID!
+            field: Int! @requiresScopes(scopes: ["entityField"])
+          }
+        `,
+        name: 'on-entity-field',
+      };
+
+      const result = composeAsFed2Subgraphs([
+        onObject,
+        onInterface,
+        onInterfaceObject,
+        onScalar,
+        onEnum,
+        onRootField,
+        onObjectField,
+        onEntityField,
+      ], { supportedFeatures });
+      assertCompositionSuccess(result);
+
+      const scopedElements = [
+        "ScopedObject",
+        "ScopedInterface",
+        "ScopedInterfaceObject",
+        "ScopedScalar",
+        "ScopedEnum",
+        "Query.scopedRootField",
+        "ObjectWithScopedField.field",
+        "EntityWithScopedField.field",
+      ];
+
+      for (const element of scopedElements) {
+        expect(
+          result.schema
+            .elementByCoordinate(element)
+            ?.hasAppliedDirective("requiresScopes")
+        ).toBeTruthy();
+      }
+    });
+
+    it('applies @requiresScopes on types as long as it is used once', () => {
+      const a1 = {
+        typeDefs: gql`
+          type Query {
+            a: A
+          }
+          type A @key(fields: "id") @requiresScopes(scopes: ["a"]) {
+            id: String!
+            a1: String
+          }
+        `,
+        name: 'a1',
+      };
+      const a2 = {
+        typeDefs: gql`
+          type A @key(fields: "id") {
+            id: String!
+            a2: String
+          }
+        `,
+        name: 'a2',
+      };
+
+      // checking composition in either order (not sure if this is necessary but
+      // it's not hurting anything)
+      const result1 = composeAsFed2Subgraphs([a1, a2], { supportedFeatures });
+      const result2 = composeAsFed2Subgraphs([a2, a1], { supportedFeatures });
+      assertCompositionSuccess(result1);
+      assertCompositionSuccess(result2);
+
+      expect(result1.schema.type('A')?.hasAppliedDirective('requiresScopes')).toBeTruthy();
+      expect(result2.schema.type('A')?.hasAppliedDirective('requiresScopes')).toBeTruthy();
+    });
+
+    it('merges @requiresScopes lists', () => {
+      const a1 = {
+        typeDefs: gql`
+          type Query {
+            a: A
+          }
+
+          type A @requiresScopes(scopes: ["a"]) @key(fields: "id") {
+            id: String!
+            a1: String
+          }
+        `,
+        name: 'a1',
+      };
+      const a2 = {
+        typeDefs: gql`
+          type A @requiresScopes(scopes: ["b"]) @key(fields: "id") {
+            id: String!
+            a2: String
+          }
+        `,
+        name: 'a2',
+      };
+
+      const result = composeAsFed2Subgraphs([a1, a2], { supportedFeatures });
+      assertCompositionSuccess(result);
+      expect(
+        result.schema.type('A')
+          ?.appliedDirectivesOf('requiresScopes')
+          ?.[0]?.arguments()?.scopes).toStrictEqual(['a', 'b']
+      );
+    });
+
+    it('merges @requiresScopes lists (intersection)', () => {
```

Comment:
change the name of this test since we don't want it to be the intersection between the lists.